Makefile

```diff
@@ -1,21 +1,25 @@
 .PHONY: start
 
 # Read values as needed from .env
+#
+# Optionally pass 'environment=<something>' to load env vars from '.env.something' instead
+# Useful to separate i.e. dev from testing databases
+#
 # If using the same variables in recipes that need to use a dotenv file other
 # than .env, remember to check that no values from .env are being used
 # inadvertently.
-ENVFILE := $(if $(environment), .env-test-e2e, .env)
+ENVFILE := $(if $(environment),.env.$(environment),.env)
 ifneq (,$(wildcard $(ENVFILE)))
     include $(ENVFILE)
+    DOCKER_COMPOSE_FILE := --env-file $(ENVFILE)
     export
 endif
 
-CIENV := $(if $(filter $(environment), ci), -f docker-compose-test-e2e.ci.yml , -f docker-compose-test-e2e.local.yml)
-API_DB_INSTANCE := $(if $(environment), test-e2e-postgresql-api, postgresql-api)
-GEO_DB_INSTANCE := $(if $(environment), test-e2e-postgresql-geo-api, postgresql-geo-api)
-REDIS_INSTANCE := $(if $(environment), test-e2e-redis, redis)
+API_DB_INSTANCE := $(if $(filter $(IS_DOCKER_E2E_TESTS_RUNTIME), true), test-e2e-postgresql-api, postgresql-api)
+GEO_DB_INSTANCE := $(if $(filter $(IS_DOCKER_E2E_TESTS_RUNTIME), true), test-e2e-postgresql-geo-api, postgresql-geo-api)
+REDIS_INSTANCE := $(if $(filter $(IS_DOCKER_E2E_TESTS_RUNTIME), true), test-e2e-redis, redis)
```

Comment:
I like the way you added the ability to manage arbitrary environments and then adapted the previous hardcoded special treatment of two environments (ci and non-ci) here, but I am not clear (no wonder, it's make-land and it's late Friday afternoon) how this new bool var is supposed to be working: I don't have it defined locally, so IIUC the _else_ branch of these conditional assignments should be used, but then e2e tests still work fine, so I am losing the plot.

Irrespective of my puzzlement, I would suggest to add a comment (in the Makefile itself, maybe, for this all to be self-contained? or in the main `README.md` to go along all/most of the other env vars we may use in `.env` files) about how to use this var and when/where to set it.